libshaderc/include/shaderc.h

```diff
@@ -24,12 +24,26 @@ extern "C" {
 #include <stdint.h>
 
 typedef enum {
-  shaderc_glsl_vertex_shader,
-  shaderc_glsl_fragment_shader,
-  shaderc_glsl_compute_shader,
-  shaderc_glsl_geometry_shader,
-  shaderc_glsl_tess_control_shader,
-  shaderc_glsl_tess_evaluation_shader
+  // Forced shader kinds. These shader kinds force the compiler to compile the
+  // source code as the specified kind of shader.
+  shaderc_glsl_forced_vertex_shader,
+  shaderc_glsl_forced_fragment_shader,
+  shaderc_glsl_forced_compute_shader,
+  shaderc_glsl_forced_geometry_shader,
+  shaderc_glsl_forced_tess_control_shader,
+  shaderc_glsl_forced_tess_evaluation_shader,
+  // Deduce the shader kind from #pragma annotation in the source code. Compiler
+  // will emit error if #pragma annotation is not found.
+  shaderc_glsl_deduce_from_pragma,
```

Comment:
With *_infer_from_source you get extra flexibility in case you want to support other languages that have nicer ways of specifying shader stage than pragma.
Set expectations about what is inferrable via documentation.?